notifications_utils/recipients.py

```diff
@@ -341,13 +339,42 @@ def validate_and_format_phone_number(number, human_readable=False):
 
 
 def validate_email_address(email_address):
-    if not re.match(email_regex, email_address) or '..' in email_address:
-        raise InvalidEmailError('Not a valid email address')
+    # almost exactly the same as by https://github.com/wtforms/wtforms/blob/master/wtforms/validators.py,
+    # with minor tweaks for SES compatibility - to avoid complications we are a lot stricter with the local part
+    # than neccessary - we don't allow any double quotes or semicolons to prevent SES Technical Failures
+    email_address = email_address.strip()
+    match = re.match(email_regex, email_address)
+
+    # not an email
+    if not match:
+        raise InvalidEmailError
+
+    hostname = match.group(1)
+    # don't allow consecutive periods in domain names
+    if '..' in hostname:
+        raise InvalidEmailError
+
+    # idna = "Internationalized domain name" - this encode/decode cycle converts unicode into its accurate ascii
+    # representation as the web uses. '例え.テスト'.encode('idna') == b'xn--r8jz45g.xn--zckzah'
+    hostname = hostname.encode('idna').decode('ascii')
+    parts = hostname.split('.')
+
+    if len(hostname) > 253 or len(parts) < 2:
+        raise InvalidEmailError
+
+    for part in parts:
+        if not part or len(part) > 63 or not hostname_part.match(part):
+            raise InvalidEmailError
+
+    # if the part after the last . is not a valid TLD then bail out
+    if not tld_part.match(parts[-1]):
+        raise InvalidEmailError
+
     return email_address
 
 
 def format_email_address(email_address):
-    return email_address.strip().lower()
+    return email_address.lower()
```

Comment:
Any reason why we no longer want to strip, seems super strict to say an email like: ' someone@mail.com' is invalid only because of the leading whitespace?